ui_testing/testcases/basic_tests/test08_contacts.py

```diff
@@ -0,0 +1,382 @@
+
+        self.contact_page.get_contact_persons_page()
+        self.base_selenium.LOGGER.info('get contact person data after refresh to make sure it was saved correctly')
+        
+        contact_persons_after_refresh = self.contact_page.get_contact_persons_data()
+        if count_of_contact_person_before_delete == 1:
+            if self.contact_page.check_contact_persons_table_is_empty():
+                self.base_selenium.LOGGER.info('contact person deleted successfully')
+            else:
+                self.base_selenium.LOGGER.info('contact person was not deleted successfully, report a bug')
+                self.assertEqual(True, False)
+
+        if len(contact_persons_after_refresh) == len(contact_persons_after_remove):
+            if self.contact_page.compare_contact_persons_data(data_after_save=contact_persons_after_refresh, data_before_save=contact_persons_after_remove):
+                self.base_selenium.LOGGER.info('contact persons have been saved successfully')
+            else:
+                self.base_selenium.LOGGER.info('contact persons was not saved successfully, you should report a BUG')
+                self.assertEqual(True, False)
+        else:
+            self.base_selenium.LOGGER.info('Data were not correctly saved, report a bug')
+            self.assertEqual(True, False)
+
+
+    def test_09_delete_contact_used_in_other_data(self):
+        """
+        New: Contact: Delete Approach: I can't delete any contact if this contact related to some data 
+        I can't delete any contact if this contact related to some data 
+        """
+
+        self.base_selenium.LOGGER.info('get orders page to get any record\'s contact to make sure that we are deleting contact that is sued in other data')
+        self.order_page.get_orders_page()
+        random_order_record = self.order_page.get_random_order_row()
+        order_data = self.base_selenium.get_row_cells_dict_related_to_header(row=random_order_record)
+        contact_name = order_data['Contact Name'].split(', ')[0]
+
+        self.contact_page.get_contacts_page()
+        self.base_selenium.LOGGER.info('filter by contact name: {}'.format(contact_name))
+        contact_record = self.contact_page.search(value=contact_name)[0]
+        if self.contact_page.check_if_table_is_empty():
+            self.base_selenium.LOGGER.info('Contact "{}" doesn\'t exist in active table'.format(contact_name))
+        else:
+            self.contact_page.click_check_box(source=contact_record)
+            self.contact_page.archive_selected_items()
+        
+        self.base_selenium.LOGGER.info('get the archived contacts')
+        self.contact_page.get_archived_contacts()
+
+        contact_archived_records = self.contact_page.search(value=contact_name)[0]
+        if self.contact_page.check_if_table_is_empty():
+            self.base_selenium.LOGGER.info('Contact is not in archived section, report a BUG')
+            self.assertEqual(True, False)
+        else:
```

Comment:
replace with  and u can add message in case of the assertion failed
```
self.assertTrue(self.contact_page.check_if_table_is_empty())
```

Reply:
done